### PR TITLE
testing: migrate test_sshutil to pytest

### PR DESCRIPTION
**Pull request by aciba90**
## Proposed Commit Message

```
testing: migrate test_sshutil to pytest
```

## Checklist:

 - [x] My code follows the process laid out in [the documentation](https://cloudinit.readthedocs.io/en/latest/topics/contributing.html)
 - [x] I have updated or added any unit tests accordingly
 - [ ] I have updated or added any documentation accordingly

### CODE REVIEW

File: tests/unittests/test_ssh_util.py [hunk 1]
```diff
@@ -322,66 +318,40 @@ def mock_getpwnam(users, username):
...
+        line = " ".join(line_args)
+
+        key = ssh_util.AuthKeyLineParser().parse(line)
+
+        assert key.base64 == content
+        assert key.keytype == ktype
+        if with_options:
+            assert key.options == options
+        else:
+            assert not key.options
```
Comment (blackboxsw) [inline tests/unittests/test_ssh_util.py:350]: More strict is 
```suggestion
            assert key.options is None
```

File: tests/unittests/test_ssh_util.py [hunk 2]
```diff
@@ -322,66 +318,40 @@ def mock_getpwnam(users, username):
...
+        if with_comment:
+            assert key.comment == comment
+        else:
+            assert not key.comment
```
Comment (blackboxsw) [inline tests/unittests/test_ssh_util.py:354]: More strict is 
```suggestion
            assert key.comment == ""
```

**Review by blackboxsw — CHANGES_REQUESTED** [review]
Minor nit and then +1

File: tests/unittests/test_ssh_util.py (hunk 1 above)
Comment (aciba90) [reply tests/unittests/test_ssh_util.py]: done

File: tests/unittests/test_ssh_util.py (hunk 2 above)
Comment (aciba90) [reply tests/unittests/test_ssh_util.py]: done

**Comment by aciba90** [comment]
> Minor nit and then +1

@blackboxsw, fixed, thanks!

**Review by blackboxsw — APPROVED** [review]
LGTM thanks

**Pull request merged**